front/lib/api/assistant/actions/process.ts

```diff
@@ -142,383 +144,376 @@ export class ProcessAction extends BaseAction {
+  ): Promise<Result<AgentActionSpecification, Error>> {
+    const owner = auth.workspace();
+    if (!owner) {
+      throw new Error("Unexpected unauthenticated call to `runRetrieval`");
+    }
 
-  return {
-    name,
-    description,
-    inputs,
-  };
-}
+    const { actionConfiguration } = this;
 
-// Generates the action specification for generation of rawInputs passed to `runProcess`.
-export async function generateProcessSpecification(
-  auth: Authenticator,
-  {
-    actionConfiguration,
-    name = "process_data_sources",
-    description,
-  }: {
-    actionConfiguration: ProcessConfigurationType;
-    name?: string;
-    description?: string;
-  }
-): Promise<Result<AgentActionSpecification, Error>> {
-  const owner = auth.workspace();
-  if (!owner) {
-    throw new Error("Unexpected unauthenticated call to `runRetrieval`");
+    const spec = await processActionSpecification({
+      actionConfiguration,
+      name,
+      description:
+        description ??
+        "Process data sources specified by the user over a specific time-frame by extracting" +
+          " structured blobs of information (complying to a fixed schema).",
+    });
+    return new Ok(spec);
   }
 
-  const spec = await processActionSpecification({
-    actionConfiguration,
-    name,
-    description:
-      description ??
-      "Process data sources specified by the user over a specific time-frame by extracting" +
-        " structured blobs of information (complying to a fixed schema).",
-  });
-  return new Ok(spec);
-}
+  // This method is in charge of running the retrieval and creating an AgentProcessAction object in
+  // the database. It does not create any generic model related to the conversation. It is possible
+  // for an AgentProcessAction to be stored (once the query params are infered) but for its execution
+  // to fail, in which case an error event will be emitted and the AgentProcessAction won't have any
+  // outputs associated. The error is expected to be stored by the caller on the parent agent message.
+  async *run(
+    auth: Authenticator,
+    {
+      agentConfiguration,
+      conversation,
+      userMessage,
+      agentMessage,
+      rawInputs,
+      functionCallId,
+      step,
+    }: BaseActionRunParams & {
+      userMessage: UserMessageType;
+    }
+  ): AsyncGenerator<
+    ProcessParamsEvent | ProcessSuccessEvent | ProcessErrorEvent,
+    void
+  > {
+    const owner = auth.workspace();
+    if (!owner) {
+      throw new Error("Unexpected unauthenticated call to `process`");
+    }
 
-/**
- * Action rendering.
- */
+    const { actionConfiguration } = this;
 
-// Internal interface for the retrieval and rendering of a actions from AgentMessage ModelIds. This
-// should not be used outside of api/assistant. We allow a ModelId interface here because for
-// optimization purposes to avoid duplicating DB requests while having clear action specific code.
-export async function processActionTypesFromAgentMessageIds(
```

Comment:
I'm assuming this was moved without changes below in the file

Reply:
Correct